compiler/src/yul/mappers/expressions.rs

```diff
@@ -459,15 +459,18 @@ fn expr_attribute_self(
 }
 
 /// Converts a storage nonce into a pointer based on the keccak256 hash
+///
+/// Pointers created here have the last byte set to zero. This is to ensure that
+/// our byte pointer sits at the start of a word (32 | `ptr` ).
 pub fn nonce_to_ptr(nonce: usize) -> yul::Expression {
-    let ptr = get_full_signature(nonce.to_string().as_bytes());
+    // set the last byte to `0x00` to ensure our pointer sits at the start of a word
```

Comment:
I'm having trouble to understand this. Let's say the full keccak of my `nonce`  would be `0xfe615a44e52a090747a78488bb7f44500a00175bbb8ff698965c2bf8ff06aeba`. As I understand it, we simply set the last byte to `0` so instead our pointer becomes `0xfe615a44e52a090747a78488bb7f44500a00175bbb8ff698965c2bf8ff060000`. I don't see how this changes anything.

>to ensure our pointer sits at the start of a word

I don't make the connection to this statement. Maybe I have to read on with the rest of the PR to understand it better...